File: PROGRAMACION/operaciones.py
# Jesus Ivan Esquivel Ruiz
# programa: python
# version del programa: 3.10.11
# descripcion del programa: este codigo genera suma de dos o mas numeros, resta de dos o mas numeros, multiplicacion de dos numeros y division de dos o mas numeros y si alguno no corresponde a esto mandara un error asi como si estos numeros son igualados a 0
# materia: programacion
# fecha de creacion del programa: 27/02/2025
# ultima modificacion del programa: 27/02/2025
import operaciones

def obtener_numeros():
    try:
        a = float(input("Ingrese el primer número: "))
        b = float(input("Ingrese el segundo número: "))
        return a, b
    except ValueError:
        print("⚠️ Entrada inválida. Por favor, ingrese números válidos.")
        return None, None

def obtener_varios_numeros():
    try:
        numeros = input("Ingrese los números separados por comas: ")
        lista_numeros = [float(num) for num in numeros.split(',')]
        return lista_numeros
    except ValueError:
        print("⚠️ Entrada inválida. Por favor, ingrese números válidos.")
        return None

def menu():
    print(" Calculadora de Operaciones Básicas ")
    print(" ¡Bienvenido! ¿En qué te puedo ayudar hoy?")
    print("1️⃣. Sumar dos números")
    print("2️⃣. Sumar varios números")
    print("3️⃣. Restar dos números")
    print("4️⃣. Multiplicar dos números")
    print("5️⃣. Multiplicar varios números")
    print("6️⃣. Dividir dos números")
    print("7️⃣. Salir")
    print("")

    while True:
        opcion = input("Seleccione una opción (1-7): ")

        if opcion == '1':
            a, b = obtener_numeros()
            if a is not None and b is not None:
                print(f"Resultado: {operaciones.sumar(a, b)}")
        elif opcion == '2':
            numeros = obtener_varios_numeros()
            if numeros is not None:
                print(f"Resultado: {operaciones.sumar_varios(*numeros)}")
        elif opcion == '3':
            a, b = obtener_numeros()
            if a is not None and b is not None:
                print(f"Resultado: {operaciones.restar(a, b)}")
        elif opcion == '4':
            a, b = obtener_numeros()
            if a is not None and b is not None:
                print(f"Resultado: {operaciones.multiplicar(a, b)}")
        elif opcion == '5':
            numeros = obtener_varios_numeros()
            if numeros is not None:
                print(f"Resultado: {operaciones.multiplicar_varios(*numeros)}")
        elif opcion == '6':
            a, b = obtener_numeros()
            if a is not None and b is not None:
                print(f"Resultado: {operaciones.dividir(a, b)}")
        elif opcion == '7':
            print(" ¡Hasta luego! ¡Que tengas un buen día!")
            break
        else:
            print("⚠️ Opción no válida, por favor seleccione una opción del 1 al 7.")

# Ejecutar el menú
menu()

        

File: PROGRAMACION/ejercicios/operaciones.js
// Autor: Jesús Iván Esquivel Ruiz
// Descripción: Este programa realiza suma, resta, multiplicación y división de números.
// Fecha de creación: 27/02/2025
// Última modificación: 27/02/2025

function obtenerNumeros() {
    try {
        const a = parseFloat(prompt("Ingrese el primer número: "));
        const b = parseFloat(prompt("Ingrese el segundo número: "));
        if (isNaN(a) || isNaN(b)) {
            throw new Error("Entrada inválida. Por favor, ingrese números válidos.");
        }
        return [a, b];
    } catch (error) {
        alert("⚠️ " + error.message);
        return null;
    }
}

function obtenerVariosNumeros() {
    try {
        const numeros = prompt("Ingrese los números separados por comas: ");
        const listaNumeros = numeros.split(',').map(num => {
            const n = parseFloat(num.trim());
            if (isNaN(n)) throw new Error("Entrada inválida. Por favor, ingrese números válidos.");
            return n;
        });
        return listaNumeros;
    } catch (error) {
        alert("⚠️ " + error.message);
        return null;
    }
}

function menu() {
    const operaciones = {
        sumar: (a, b) => a + b,
        sumarVarios: (...nums) => nums.reduce((sum, num) => sum + num, 0),
        restar: (a, b) => a - b,
        multiplicar: (a, b) => a * b,
        multiplicarVarios: (...nums) => nums.reduce((product, num) => product * num, 1),
        dividir: (a, b) => {
            if (b === 0) throw new Error("La división por cero no está permitida.");
            return a / b;
        }
    };

    alert("Calculadora de Operaciones Básicas\n¡Bienvenido! ¿En qué te puedo ayudar hoy?");
    let continuar = true;

    while (continuar) {
        const opcion = prompt(
            "Seleccione una opción (1-7):\n" +
            "1️⃣. Sumar dos números\n" +
            "2️⃣. Sumar varios números\n" +
            "3️⃣. Restar dos números\n" +
            "4️⃣. Multiplicar dos números\n" +
            "5️⃣. Multiplicar varios números\n" +
            "6️⃣. Dividir dos números\n" +
            "7️⃣. Salir"
        );

        try {
            switch (opcion) {
                case '1': {
                    const [a, b] = obtenerNumeros() || [];
                    if (a !== undefined && b !== undefined) {
                        alert(`Resultado: ${operaciones.sumar(a, b)}`);
                    }
                    break;
                }
                case '2': {
                    const numeros = obtenerVariosNumeros();
                    if (numeros) {
                        alert(`Resultado: ${operaciones.sumarVarios(...numeros)}`);
                    }
                    break;
                }
                case '3': {
                    const [a, b] = obtenerNumeros() || [];
                    if (a !== undefined && b !== undefined) {
                        alert(`Resultado: ${operaciones.restar(a, b)}`);
                    }
                    break;
                }
                case '4': {
                    const [a, b] = obtenerNumeros() || [];
                    if (a !== undefined && b !== undefined) {
                        alert(`Resultado: ${operaciones.multiplicar(a, b)}`);
                    }
                    break;
                }
                case '5': {
                    const numeros = obtenerVariosNumeros();
                    if (numeros) {
                        alert(`Resultado: ${operaciones.multiplicarVarios(...numeros)}`);
                    }
                    break;
                }
                case '6': {
                    const [a, b] = obtenerNumeros() || [];
                    if (a !== undefined && b !== undefined) {
                        alert(`Resultado: ${operaciones.dividir(a, b)}`);
                    }
                    break;
                }
                case '7': {
                    alert("¡Hasta luego! ¡Que tengas un buen día!");
                    continuar = false;
                    break;
                }
                default:
                    alert("⚠️ Opción no válida, por favor seleccione una opción del 1 al 7.");
            }
        } catch (error) {
            alert("⚠️ " + error.message);
        }
    }
}

// Ejecutar el menú
menu();
